core/embed/rust/src/ui/model_tr/component/homescreen.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{
+    micropython::{buffer::StrBuffer, gc::Gc},
+    storage::{get_avatar, get_avatar_len},
+    trezorhal::usb::usb_configured,
+    ui::{
+        component::{Child, Component, Event, EventCtx, Pad},
+        display::{
+            fill_background_for_text, rect_fill,
+            toif::{icon_from_toif, Toif},
+            Font, Icon,
+        },
+        event::USBEvent,
+        geometry::{self, Alignment, Offset, Point, Rect},
+    },
+};
+
+use super::{
+    super::constant, common::display_center, theme, ButtonController, ButtonControllerMsg,
+    ButtonLayout,
+};
+
+const AREA: Rect = constant::screen();
+const TOP_CENTER: Point = AREA.top_center();
+const LABEL_Y: i16 = constant::HEIGHT - 5;
+const LOCKED_INSTRUCTION_Y: i16 = 34;
+const LOGO_ICON_TOP_MARGIN: i16 = 12;
+const LOCK_ICON_TOP_MARGIN: i16 = 12;
+const NOTIFICATION_HEIGHT: i16 = 12;
+
+pub struct Homescreen {
+    label: StrBuffer,
+    notification: Option<(StrBuffer, u8)>,
+    pad: Pad,
+    /// Used for HTC functionality to lock device from homescreen
+    invisible_buttons: Child<ButtonController<StrBuffer>>,
+}
+
+pub enum HomescreenMsg {
+    Dismissed,
+}
+
+impl Homescreen {
+    pub fn new(label: StrBuffer, notification: Option<(StrBuffer, u8)>) -> Self {
+        // NOTE: for some reason the text cannot be empty string, it was panicking at
+        // library/core/src/str/mod.rs:107
```

Comment:
Seems to be working with empty strings now? If some piece of our code cannot handle empty strings better find out what it is.

Reply:
Right, now it works with empty string. Reverted in [51a69f7](https://github.com/trezor/trezor-firmware/pull/2610/commits/51a69f79f6a81b035f974d8a7eae82aa9872ce58)